tests/d3d12_shaders.c

```diff
@@ -15005,3 +15005,113 @@ void test_derivative_hoisting_dxil(void)
 {
     test_derivative_hoisting(true);
 }
+
+
+static void test_root_constant_indexing(bool use_dxil)
+{
+    D3D12_COMPUTE_PIPELINE_STATE_DESC pso_desc;
+    D3D12_HEAP_PROPERTIES heap_properties;
+    D3D12_RESOURCE_DESC resource_desc;
+    D3D12_ROOT_SIGNATURE_DESC rs_desc;
+    D3D12_ROOT_PARAMETER rs_args[2];
+    uint32_t constant_data[12 * 4];
+    ID3D12Resource *uav_resource;
+    struct resource_readback rb;
+    struct test_context context;
+    unsigned int i;
+    HRESULT hr;
+
+#include "shaders/shaders/headers/cs_root_constant_indexing.h"
+
+    if (!init_compute_test_context(&context))
+        return;
+
+    memset(constant_data, 0, sizeof(constant_data));
+
+    for (i = 0u; i < 12u; i++)
+        constant_data[4u * i] = 0x80000000u + i;
+
+    memset(rs_args, 0, sizeof(rs_args));
+    rs_args[0].ParameterType = D3D12_ROOT_PARAMETER_TYPE_32BIT_CONSTANTS;
+    rs_args[0].Constants.Num32BitValues = ARRAY_SIZE(constant_data);
+    rs_args[0].ShaderVisibility = D3D12_SHADER_VISIBILITY_ALL;
+
+    rs_args[1].ParameterType = D3D12_ROOT_PARAMETER_TYPE_UAV;
+    rs_args[1].ShaderVisibility = D3D12_SHADER_VISIBILITY_ALL;
+
+    memset(&rs_desc, 0, sizeof(rs_desc));
+    rs_desc.NumParameters = ARRAY_SIZE(rs_args);
+    rs_desc.pParameters = rs_args;
+
+    hr = create_root_signature(context.device, &rs_desc, &context.root_signature);
+    ok(hr == S_OK, "Failed to create root signature, hr %#x.\n", hr);
+
+    memset(&pso_desc, 0, sizeof(pso_desc));
+    pso_desc.pRootSignature = context.root_signature;
+    pso_desc.CS = use_dxil ? cs_root_constant_indexing_dxil : cs_root_constant_indexing_dxbc;
+
+    /* For DXIL, there seems to be some runtime validation going on */
+    hr = ID3D12Device_CreateComputePipelineState(context.device,
+            &pso_desc, &IID_ID3D12PipelineState, (void**)&context.pipeline_state);
+
+    if (use_dxil)
+        ok(hr == E_INVALIDARG, "Got hr %#x, expected E_INVALIDARG.\n", hr);
```

Comment:
dxil-spirv will just allow it, so hr == S_OK if running against vkd3d-proton should be fine.

Reply:
simply made this todo for now, it *does* fail on native after all.